### DISCUSSION TITLE
Relative permeability curve

### DISCUSSION THREAD

**Discussion by gxw-ll** [PMEAL/OpenPNM#2844] [Q&A]
Hi Prof,
I meet a question in the Predicting relative permeability in using openpnm.
I generated the matrix and pores using the quartet structure generating set method, extracted the pore network model using the SNOW2 algorithm, and used gas to invade water on the pore network model, resulting in a very strange relative permeability curve, the relative permeability of water is very small, where do I need to adjust the model？
Here is my relative permeability curve:
<img width="420" alt="relative" src="https://github.com/PMEAL/OpenPNM/assets/45071471/e33e6a52-a874-4f24-893e-3f17a5001092">

**Comment by jgostick** [comment]
There are many things that could be wrong.  You should try breaking it into manageable chunks.  Is your network extraction ok?  Is it's Pc curve good?  Can you get good rel perm curves with a cubic network?  What are the differences between the extracted network and the cubic one?  Etc.

**Reply by gxw-ll** [reply]
OK,，Thank you for your prompt response! It's very helpful for my current situation. I will try the directions you provided.

**Reply by gxw-ll** [reply]
Yes, based on your advice, I decided to modify the pore network in my simulation. Initially, I used a voxel size of 30 x 30 x 30, resulting in 149 pores. After the adjustment, I switched to a voxel size of 300 x 300 x 300, yielding a total of 48,872 pores. This modification proved to be instrumental as it allowed me to obtain the desired curve in my simulation results.Why does voxel size have such a significant impact?The following is a portion of my code and fig
<img width="420" alt="relative" src="https://github.com/PMEAL/OpenPNM/assets/45071471/4ca951c1-51fb-4cf1-a933-2da27e795edd">

` # Matrix,pores and hydrates are generated using the quadruple structure generation set method
    im_data = loadmat('D:/python_script/TestRelativePermeablity/testPF1.mat', squeeze_me=True) 
    im_data[im_data == 2] = 0 # Hydrate is equivalent to matrix, im_data contains 0,1,2. 0 as matrix, 1 as pore, and 2 as hydrate
    im = im_data['E'].astype(bool) #im = 300*300*300
    ps.metrics.porosity(im)
    length_x, length_y, length_z = im.shape
    shape = [length_x, length_y, length_z]
    net = ps.networks.snow2(phases = im, voxel_size=1e-3)
    pn = op.io.network_from_porespy(net.network)
    pn.regenerate_models()`

**Reply by jgostick** [reply]
You must be mixing up terminology because voxel size is a scalar, not 30x30x30.  Is 300x300x300 your image image size?  If so then 30x30x30 is much too small so that would be why your results were gibberish. 300-cubed is about the minimum image size for getting a good extraction.

**Reply by gxw-ll** [reply]
Yes, you're right,I made a mistake. Thank you for your prompt response.
